Add flag to collect from all network interfaces

Bacon41 opened the pull request:
An '--all' flag added to the network metric to allow collection from
all interfaces. Default will collect only from 'eth_' and 'bond_'.

Bacon41 committed:
Add flag to collect from all network interfaces

An '--all' flag added to the network metric to allow collection from
all interfaces. Default will collect only from 'eth*' and 'bond*'.

paulczar commented:
looks like a good start.

maybe we should include an `--interfaces` option that lists interfaces that we want to see.  could default as `eth,bond`.    can transform that into a list and use it for pattern checking interfaces ?

na-gupta commented on sensu/plugins/metrics-net.rb:
```diff
@@ -80,7 +85,7 @@ def run
       next if File.file?(iface_path)
       iface = File.basename(iface_path)
-      next if iface == 'lo'
+      next if !config[:all] and !iface.include?('eth') and !iface.include?('bond')
```
I think since the intent is to filter eth\* and bond\* and not `*eth*` and `*bond*`, start_with would be more appropriate than include.

Bacon41 replied:
Fixed

na-gupta commented on sensu/plugins/metrics-net.rb:
```diff
@@ -80,7 +85,7 @@ def run
     Dir.glob('/sys/class/net/*').each do |iface_path|
       next if File.file?(iface_path)
       iface = File.basename(iface_path)
-      next if iface == 'lo'
```
We might want to filter out 'lo' even when the all flag is set, depending on the requirements

Bacon41 replied:
Fixed

Bacon41 committed:
Add option to specify which interfaces to collect from

An '--interfaces' option added to the network metric to allow greater
control over which interfaces will be collected from. Defaults to
'eth' and 'bond'. 'lo' always skipped.

Bacon41 commented:
Interfaces option added.

na-gupta commented:
Looks good to me.
